### PR TITLE
Python Bindings: Use python-pip for build and installation

### PR DESCRIPTION
**Pull request by rcrdnalor**
This pull-request consists of two parts:
 - Cleanup build of the Python Bindings
 - Use python-pip for build and installation of the bindings
   for python >= 3.11 and python-pip >= 23.0.1
 
This PR removes the deprecation warning:
"""SetuptoolsDeprecationWarning: setup.py install is deprecated. 
Use build and pip and other standards-based tools."""
It implements compatibility to PEP 517 and PEP 688,
which generally means "A python compatible build and install,
compatible to the rules defined by the OS-distribution."
See issue #731 for details.
 
See https://peps.python.org/pep-0517/
and https://peps.python.org/pep-0668/ 
 
The second part needs an update of MythTV/packaging, too.
A separate pull-request is in preparation.

Both PRs need to be applied at the same time, otherwise
the Launchpad build fails for all builds.
Reason: The mythtv/bindings/python/Makefile is patched
for Ubuntu packaging.

##### Checklist

- [x] contribution does not duplicate one of our [existing pull requests](https://github.com/MythTV/mythtv/pulls)
- [x] contribution is in a branch rebased against [master](https://github.com/MythTV/mythtv)
- [x] code compiles successfully without errors
- [x] code follows the [MythTV Coding Standards](https://www.mythtv.org/wiki/Coding_Standards)
- [x] documentation added/updated/removed where necessary
- [x] commits are logically organised and have [good commit messages](https://chris.beams.io/posts/git-commit)

### CODE REVIEW

**Review by paul-h — APPROVED** [review]
This is not an area of python I'm at all familiar with but at a glance the patch looks OK to me.

Is this required for 23.04? We may be to late for it to be included?

**Comment by billmeek** [comment]
Roland, can you see the same 'SyntaxError: invalid syntax' in _versions.py
when doing a make install? I did the test as a patch, not a merge since there
have been some changes to the bindings lately.

Sorry for my delay, can you rebase the PR.

```
make -C bindings/python install INSTALL_ROOT=
make[1]: Entering directory '/home/bill/source/mythtv/mythtv/bindings/python'
python3 setup.py install --skip-build --root="/" 
running install
/usr/local/lib/python3.10/dist-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
running install_lib
byte-compiling /usr/local/lib/python3.10/dist-packages/MythTV/_versions.py to _versions.cpython-310.pyc
  File "usr/local/lib/python3.10/dist-packages/MythTV/_versions.py", line 16
    INSTALL_PREFIX = /usr/local
                     ^
SyntaxError: invalid syntax

running install_egg_info
running egg_info
```
Causes this when I try something using the bindings:
```
Traceback (most recent call last):
  File "/home/bill/bin/up", line 10, in <module>
    from MythTV import (MythBE, exceptions)
  File "/usr/local/lib/python3.10/dist-packages/MythTV/__init__.py", line 32, in <module>
    from . import _versions
  File "/usr/local/lib/python3.10/dist-packages/MythTV/_versions.py", line 16
    INSTALL_PREFIX = /usr/local
```

**Comment by rcrdnalor** [comment]
Yes, my bad. In the file "_versions.py.in" this line should be changed:
INSTALL_PREFIX = @MYTHTV_INSTALL_PREFIX@
to
INSTALL_PREFIX = '@MYTHTV_INSTALL_PREFIX@'
Note the missing single quotes.
I do not know how I could overlooked that.
Thank you for pointing this out.
Roland

**Comment by billmeek** [comment]
@rcrdnalor, built on 2 Ubuntu 22.04 hosts with the above and ran some Python tools I use. All work fine.

I'll note that running ./configure is required.

**Comment by rcrdnalor** [comment]
@billmeek 
Bill,
thank you for testing it.
The big changes are related to Debian/testing (bookworm) or Ubuntu Lunar (23.04), see #731,
They will be needed at least in Ubuntu 24.04, see 
https://github.com/rcrdnalor/mythtv/blob/e768a78acd1f2c06ff40f33af7d610f27cff1ea8/mythtv/configure#L6579
You need a VM to test this, see https://github.com/MythTV/mythtv/issues/731 for the rationale.
This PR uses the package 'pythton3-pip' to install the locally build of the MythTV-Python package.

Regards,
Roland

P.s.: I did a mistake on my my local PR branch. Thus I can't rebase it by now.
I apologize for your additional effort.

**Comment by rcrdnalor** [comment]
@linuxdude42 
This is the most complex change I ever done for MythTV, because it needs to be aligned with https://github.com/MythTV/packaging/pull/153
Could you please check if these changes are aligned to the way of your release scripting chain?
The proposed changes use now a separate file "_versions.py" which is created by ./configure.

File: mythtv/bindings/python/MythTV/__init__.py [hunk 1]
```diff
@@ -42,6 +43,6 @@
 from .methodheap import *
 
 
-__version__ = OWN_VERSION
+__version__ = ".".join(str(x) for x in _versions.OWN_VERSION)
```
Comment (linuxdude42) [inline mythtv/bindings/python/MythTV/__init__.py:46]: MYTHTV_PYTHON_OWN_VERSION is unique to the python bindings. If it doesn't have to be a list in _versions.py, you could change configure to write the string '34.0.-1.0' and not have to manipulate it here.

**Review by linuxdude42 — APPROVED** [review]
Looks good. Moving the whole file static.py to static.py.in bothered me, so I really like moving it back and having the separate _versions.py file.

File: mythtv/bindings/python/MythTV/__init__.py (hunk 1 above)
Comment (rcrdnalor) [reply mythtv/bindings/python/MythTV/__init__.py:46]: The defined `OWN_VERSION` is kept for legacy reasons. I've seen some script that uses it.

**Pull request merged**